### PR TITLE
Add 'ignore' option

### PR DESCRIPTION
**Pull request by bezoerb**
Adresses #58

### CODE REVIEW

**Comment by rupl** [comment]
Wow nice! I will give this a spin tomorrow on a couple sites and see how it works.

**Comment by bezoerb** [comment]
looking forward to some feedback ;)

**Comment by rupl** [comment]
I dropped this into a couple projects and all my functional tests worked brilliantly!

I'm trying to get a branch going that ignores specific values (e.g. `url`) but I haven't nailed it yet.

**Comment by bezoerb** [comment]
Filtering values is not supported yet by filter-css.
I planed to support this but I'm not quite sure about the syntax yet

**Comment by rupl** [comment]
whoops yeah the branch I mentioned was for filter-css, not critical. The version in this PR works great for selectors and other blocks like `@font-face`

**Comment by bezoerb** [comment]
@rupl Added declaration matching. Wanna give it another try?

**Comment by bezoerb** [comment]
@rupl ping ;)

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "cheerio": "^0.19.0",
     "clean-css": "^3.1.9",
     "cli": "^0.6.6",
+    "filter-css": "0.0.1",
```
Comment (rupl) [inline package.json]: seems like a dupe dependency here? it told me it was opting for `0.0.1` when I installed

**Comment by rupl** [comment]
The type/declarations ignores seem to be mutually exclusive. Using the `filter-css` docs I was able to either filter a block like `@font-face` OR filter out a declaration like `url`

Here is the syntax that made the declarations work:

```js
{
  ignore: {declarations: [/background-image/, /src/]}
}
```

The syntax for `@font-face` was as follows:

```js
{
  ignore: ['@font-face']
}
```

So I couldn't figure out how to combine them. I tried something like this:

```js
{
  ignore: [{declarations: [/background-image/, /src/]}, '@font-face']
}
```

**Comment by bezoerb** [comment]
@rupl it seems the interface of `filter-css` isn't optimal yet ;)
You could have achieved this with the following:
```js
ignore: {
  declarations: [/background-image/, /src/],
  types: ['@font-face']
}
```
Maybe you would like to help me making it better?
https://github.com/bezoerb/filter-css/issues/1

**Comment by rupl** [comment]
ah ha, that makes sense now that I see it. I'm happy to drop some thoughts in the other issue.

**Comment by rupl** [comment]
:tada:

**Comment by bezoerb** [comment]
@rupl thanks for the support :)

**Pull request merged**